### PR TITLE
Spring batch tests to java - basic

### PR DESCRIPTION
**Pull request by zeitlinger**
This is the first part of https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12004 to make it smaller

### CODE REVIEW

**Comment by zeitlinger** [comment]
@steverao @jaydeluca can you take a look?

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/JavaConfigBatchJobTest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.basic;
+
+import io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.runner.ApplicationConfigRunner;
+import io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.runner.SpringBatchApplication;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+public class JavaConfigBatchJobTest extends SpringBatchTest {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/JavaConfigBatchJobTest.java:13]:
```suggestion
class JavaConfigBatchJobTest extends SpringBatchTest {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/JsrConfigBatchJobTest.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.basic;
+
+import io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.runner.JavaxBatchConfigRunner;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class JsrConfigBatchJobTest extends SpringBatchTest {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/JsrConfigBatchJobTest.java:11]:
```suggestion
class JsrConfigBatchJobTest extends SpringBatchTest {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("BatchJob taskletJob")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasAttribute(AttributeKey.stringKey("job.system"), "spring_batch"),
+                span ->
+                    span.hasName("BatchJob taskletJob.step")
+                        .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:51]:
```suggestion
                        .hasKind(SpanKind.INTERNAL)
                        .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.hasName("BatchJob taskletJob.step.Tasklet")
+                        .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:55]:
```suggestion
                        .hasKind(SpanKind.INTERNAL)
                        .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("BatchJob taskletJob")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasAttribute(AttributeKey.stringKey("job.system"), "spring_batch"),
+                span ->
+                    span.hasName("BatchJob taskletJob.step")
+                        .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:72]:
```suggestion
                        .hasKind(SpanKind.INTERNAL)
                        .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+                assertThat(spanData.getName())
+                    .matches("BatchJob partitionedJob.partitionWorkerStep:partition[01].Chunk"))
+        .hasParent(trace.getSpan(index));
+  }
+
+  protected boolean hasPartitionManagerStep() {
+    return true;
+  }
+
+  // should be moved to SpanDataAssert
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:269]: what does this comment mean?

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+  private static void verifyException(SpanDataAssert span, Throwable exception) {
+    span.hasStatus(StatusData.error())
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:271]:
```suggestion
    span.hasStatus(StatusData.error())
      .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+  @Test
+  void shouldTraceChunkedItemsJob() {
+    runner.runJob("itemsAndTaskletJob");
+
+    testing.waitAndAssertTraces(
+        trace -> {
+          Consumer<SpanDataAssert> chunk =
+              span ->
+                  span.hasName("BatchJob itemsAndTaskletJob.itemStep.Chunk")
+                      .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:92]:
```suggestion
                      .hasKind(SpanKind.INTERNAL)
                      .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+                      .hasParent(trace.getSpan(1));
+          trace.hasSpansSatisfyingExactly(
+              span ->
+                  span.hasName("BatchJob itemsAndTaskletJob")
+                      .hasKind(SpanKind.INTERNAL)
+                      .hasAttribute(AttributeKey.stringKey("job.system"), "spring_batch"),
+              span ->
+                  span.hasName("BatchJob itemsAndTaskletJob.itemStep")
+                      .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:101]:
```suggestion
                      .hasKind(SpanKind.INTERNAL)
                      .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 10]
```diff
@@ -0,0 +1,283 @@
...
+                      .hasParent(trace.getSpan(0)),
+              chunk,
+              chunk,
+              chunk,
+              span ->
+                  span.hasName("BatchJob itemsAndTaskletJob.taskletStep")
+                      .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:108]:
```suggestion
                      .hasKind(SpanKind.INTERNAL)
                      .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 11]
```diff
@@ -0,0 +1,283 @@
...
+                      .hasParent(trace.getSpan(0)),
+              span ->
+                  span.hasName("BatchJob itemsAndTaskletJob.taskletStep.Tasklet")
+                      .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:112]:
```suggestion
                      .hasKind(SpanKind.INTERNAL)
                      .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 12]
```diff
@@ -0,0 +1,283 @@
...
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("BatchJob flowJob")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasAttribute(AttributeKey.stringKey("job.system"), "spring_batch"),
+                span ->
+                    span.hasName("BatchJob flowJob.flowStep1")
+                        .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:130]:
```suggestion
                        .hasKind(SpanKind.INTERNAL)
                        .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 13]
```diff
@@ -0,0 +1,283 @@
...
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.hasName("BatchJob flowJob.flowStep1.Tasklet")
+                        .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:134]:
```suggestion
                        .hasKind(SpanKind.INTERNAL)
                        .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 14]
```diff
@@ -0,0 +1,283 @@
...
+                        .hasParent(trace.getSpan(1)),
+                span ->
+                    span.hasName("BatchJob flowJob.flowStep2")
+                        .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:138]:
```suggestion
                        .hasKind(SpanKind.INTERNAL)
                        .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 15]
```diff
@@ -0,0 +1,283 @@
...
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.hasName("BatchJob flowJob.flowStep2.Tasklet")
+                        .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:142]:
```suggestion
                        .hasKind(SpanKind.INTERNAL)
                        .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 16]
```diff
@@ -0,0 +1,283 @@
...
+                                assertThat(spanData.getName())
+                                    .matches("BatchJob splitJob.splitFlowStep[12]"))
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.satisfies(
+                            spanData ->
+                                assertThat(spanData.getName())
+                                    .matches("BatchJob splitJob.splitFlowStep[12].Tasklet"))
+                        .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:169]:
```suggestion
                        .hasKind(SpanKind.INTERNAL)
                        .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 17]
```diff
@@ -0,0 +1,283 @@
...
+                        .hasParent(trace.getSpan(1)),
+                span ->
+                    span.satisfies(
+                            spanData ->
+                                assertThat(spanData.getName())
+                                    .matches("BatchJob splitJob.splitFlowStep[12]"))
+                        .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:176]:
```suggestion
                        .hasKind(SpanKind.INTERNAL)
                        .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 18]
```diff
@@ -0,0 +1,283 @@
...
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.satisfies(
+                            spanData ->
+                                assertThat(spanData.getName())
+                                    .matches("BatchJob splitJob.splitFlowStep[12].Tasklet"))
+                        .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:183]:
```suggestion
                        .hasKind(SpanKind.INTERNAL)
                        .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 19]
```diff
@@ -0,0 +1,283 @@
...
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("BatchJob decisionJob")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasAttribute(AttributeKey.stringKey("job.system"), "spring_batch"),
+                span ->
+                    span.hasName("BatchJob decisionJob.decisionStepStart")
+                        .hasKind(SpanKind.INTERNAL)
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:200]: this 0 attribute assertion is missing for many others as well
```suggestion
                        .hasKind(SpanKind.INTERNAL)
                        .hasTotalAttributeCount(0)
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java [hunk 20]
```diff
@@ -0,0 +1,283 @@
...
+                        .hasParent(trace.getSpan(3))));
+  }
+
+  @Test
+  void shouldTracePartitionedJob() {
+    runner.runJob("partitionedJob");
+
+    testing.waitAndAssertTraces(
+        trace -> {
+          int index = 2;
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:222]: this is only referenced twice, is there a reason we need to declare it as a var?

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/XmlConfigBatchJobTest.java [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.basic;
+
+import io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.runner.ApplicationConfigRunner;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+public class XmlConfigBatchJobTest extends SpringBatchTest {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/XmlConfigBatchJobTest.java:12]:
```suggestion
class XmlConfigBatchJobTest extends SpringBatchTest {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventChunkListener.java [hunk 22]
```diff
@@ -0,0 +1,26 @@
...
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.jsr;
+
+import io.opentelemetry.api.trace.Span;
+import javax.batch.api.chunk.listener.ChunkListener;
+
+public class CustomEventChunkListener implements ChunkListener {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventChunkListener.java:11]:
```suggestion
class CustomEventChunkListener implements ChunkListener {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventItemProcessListener.java [hunk 23]
```diff
@@ -0,0 +1,26 @@
...
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.jsr;
+
+import io.opentelemetry.api.trace.Span;
+import javax.batch.api.chunk.listener.ItemProcessListener;
+
+public class CustomEventItemProcessListener implements ItemProcessListener {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventItemProcessListener.java:11]:
```suggestion
class CustomEventItemProcessListener implements ItemProcessListener {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventItemReadListener.java [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.jsr;
+
+import io.opentelemetry.api.trace.Span;
+import javax.batch.api.chunk.listener.ItemReadListener;
+
+public class CustomEventItemReadListener implements ItemReadListener {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventItemReadListener.java:11]:
```suggestion
class CustomEventItemReadListener implements ItemReadListener {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventItemWriteListener.java [hunk 25]
```diff
@@ -0,0 +1,27 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.jsr;
+
+import io.opentelemetry.api.trace.Span;
+import java.util.List;
+import javax.batch.api.chunk.listener.ItemWriteListener;
+
+public class CustomEventItemWriteListener implements ItemWriteListener {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventItemWriteListener.java:12]:
```suggestion
class CustomEventItemWriteListener implements ItemWriteListener {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventJobListener.java [hunk 26]
```diff
@@ -0,0 +1,21 @@
...
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.jsr;
+
+import io.opentelemetry.api.trace.Span;
+import javax.batch.api.listener.JobListener;
+
+public class CustomEventJobListener implements JobListener {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventJobListener.java:11]:
```suggestion
class CustomEventJobListener implements JobListener {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventStepListener.java [hunk 27]
```diff
@@ -0,0 +1,21 @@
...
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.jsr;
+
+import io.opentelemetry.api.trace.Span;
+import javax.batch.api.listener.StepListener;
+
+public class CustomEventStepListener implements StepListener {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventStepListener.java:11]:
```suggestion
class CustomEventStepListener implements StepListener {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/SingleItemReader.java [hunk 28]
```diff
@@ -0,0 +1,34 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.jsr;
+
+import java.io.Serializable;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.batch.api.chunk.ItemReader;
+
+public class SingleItemReader implements ItemReader {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/SingleItemReader.java:12]:
```suggestion
class SingleItemReader implements ItemReader {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestBatchlet.java [hunk 29]
```diff
@@ -0,0 +1,36 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.jsr;
+
+import javax.batch.api.BatchProperty;
+import javax.batch.api.Batchlet;
+import javax.inject.Inject;
+
+public class TestBatchlet implements Batchlet {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestBatchlet.java:12]:
```suggestion
class TestBatchlet implements Batchlet {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestDecider.java [hunk 30]
```diff
@@ -0,0 +1,16 @@
...
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.jsr;
+
+import javax.batch.api.Decider;
+import javax.batch.runtime.StepExecution;
+
+public class TestDecider implements Decider {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestDecider.java:11]:
```suggestion
class TestDecider implements Decider {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestItemProcessor.java [hunk 31]
```diff
@@ -0,0 +1,16 @@
...
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.jsr;
+
+import javax.batch.api.chunk.ItemProcessor;
+import org.codehaus.groovy.runtime.DefaultGroovyMethods;
+
+public class TestItemProcessor implements ItemProcessor {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestItemProcessor.java:11]:
```suggestion
class TestItemProcessor implements ItemProcessor {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestItemReader.java [hunk 32]
```diff
@@ -0,0 +1,43 @@
...
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.jsr;
+
+import java.io.Serializable;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.batch.api.chunk.ItemReader;
+
+public class TestItemReader implements ItemReader {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestItemReader.java:15]:
```suggestion
class TestItemReader implements ItemReader {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestItemWriter.java [hunk 33]
```diff
@@ -0,0 +1,34 @@
...
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.jsr;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import javax.batch.api.chunk.ItemWriter;
+import org.codehaus.groovy.runtime.DefaultGroovyMethods;
+
+public class TestItemWriter implements ItemWriter {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestItemWriter.java:14]:
```suggestion
class TestItemWriter implements ItemWriter {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestPartitionedItemReader.java [hunk 34]
```diff
@@ -0,0 +1,79 @@
...
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.jsr;
+
+import java.io.Serializable;
+import javax.batch.api.BatchProperty;
+import javax.batch.api.chunk.ItemReader;
+import javax.inject.Inject;
+
+public class TestPartitionedItemReader implements ItemReader {
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestPartitionedItemReader.java:13]:
```suggestion
class TestPartitionedItemReader implements ItemReader {
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestItemWriter.java [hunk 35]
```diff
@@ -0,0 +1,34 @@
...
+      this.items.add(DefaultGroovyMethods.asType(item, Integer.class));
+    }
+  }
+
+  @Override
+  public Serializable checkpointInfo() {
+    return null;
+  }
+
+  private final List<Integer> items = new ArrayList<>();
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/TestItemWriter.java:33]: for many of these classes the fields should [go at the top](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/style-guideline.md#ordering-of-class-contents)

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java (hunk 6 above)
Comment (zeitlinger) [reply instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java:284]: oh - just saw that this method already exists: `hasException`

**Comment by zeitlinger** [comment]
@jaydeluca please check again

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/runner/JavaxBatchConfigRunner.java [hunk 36]
```diff
@@ -0,0 +1,39 @@
...
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.runner;
+
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
```
Comment (jaydeluca) [inline instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/runner/JavaxBatchConfigRunner.java:11]: duplicate comment 

```suggestion
```

File: instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventChunkListener.java (hunk 22 above)
Comment (jaydeluca) [reply instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventChunkListener.java:11]: theres still a lot of public classes, do they need to be public or could they be package private?

Comment (zeitlinger) [reply instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/jsr/CustomEventChunkListener.java:11]: yes, the remaining public classes are loaded via reflection

**Review by jaydeluca — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Pull request merged**